pkg/storage/export/export/export.go

```diff
@@ -508,17 +508,22 @@ func (ctrl *VMExportController) updateVMExport(vmExport *exportv1.VirtualMachine
 		populateInitialVMExportStatus(vmExport)
 	}
 
-	if err := ctrl.handleVMExportToken(vmExport); err != nil {
-		return 0, err
-	}
-
 	if ctrl.isSourcePvc(&vmExport.Spec) {
+		if err := ctrl.handleVMExportToken(vmExport, ctrl.getPVCFromSourcePVC); err != nil {
```

Comment:
Would it make sense to move calling `handleVMExportToken` to `hanleSource()`?

Reply:
It does, let me do that.